### ISSUE TITLE
GitJournal deletes other yaml metadata not related to it.

### ISSUE THREAD

**Issue by Uzay-G** [GitJournal/GitJournal#374]
## Description

Hi, I've been trying to integrate GitJournal for use with [Archivy](https://archivy.github.io) and its [git extension](https://github.com/archivy/archivy-git).

However, archivy also uses yaml frontmatter with some attributes like `path`, `type` and `desc`. GitJournal seems to delete these `path` and `type` attributes when I make an edit to a note.

Would it be possible for it to not behave like this?

**Comment by vHanda** [comment]
Hi. I can't seem to reproduce this bug.

I've tried to create a new note with some custom frontmatter, and when editing the note the frontmatter is preserved. Could you please share an example?

![Screenshot_1609062274](https://user-images.githubusercontent.com/426467/103168094-ae031600-4830-11eb-9a9b-1d41a3eb144e.png)

**Comment by Uzay-G** [comment]
Oh ok I see. Try adding a `date` attribute for example.

Once you save the edit, you'll see it has disappeared.

Maybe this attribute is linked to something internal for gitjournal?

**Comment by vHanda** [comment]
Yup. With 'date' I managed to reproduce it. This shouldn't be happening.

**Comment by dleeftink** [comment]
Can confirm the same behaviour for YAML key `type` using GitJournal for Android 1.75.1+2324

**Comment by dleeftink** [comment]
> Can confirm the same behaviour for YAML key `type` using GitJournal for Android 1.75.1+2324

In addition, single quotes are deleted from YAML, which is problematic when trying to preserve a key with whitespace like so:

```
---
_title: ' TitleStartingWithWhitespace'
         ^whitespace
---
```

After editing becomes
```
---
_title: TitleStartingWithWhitespace
        ^no quotes, no whitespace
---
```

**Comment by craignicol** [comment]
Just had this issue myself. See this git diff for example. I have a Trello importer that added all the front matter here, but GitJournal scrubbed it all and only left me with "modified". I believe GitJournal doesn't recognise the timezone either as the note appears at the bottom when sorted by modified date even though I have notes from 2019 in there.

 ```
@@ -1,14 +1,5 @@
---
- created: 2021-03-27T10:28:48.350Z
- modified: 2021-03-27T10:28:48.350Z
- title: Creating a Decision Journal: Template And Example Included
- source: Trello
- tags: 
- references: 
- board: Training
- list: Project = This Year Career
- published: false
- trello-url: https://trello.com/c/znhStEq9
+ modified: 2021-07-02T19:45:10+01:00
---
```

**Comment by craignicol** [comment]
Actually, mine might be a slightly different issue. I found all the affected notes, and they all have a : character in the title, which I guess breaks the yaml parser? So all the yaml headers are lost.

**Comment by vHanda** [comment]
Hello. Dart unfortunately doesn't ship with a yaml serializer, and I landed up writing a simple one, which has lots of limitations.

If you can post clear examples which are reproducible, then I can start fixing them.

Cheers.

**Comment by craignicol** [comment]
I've written a bare bones yaml parser for some of this myself, and the key issue seems to be : or - signs in the value. 

YAML spec allows anything in the value, so I think splitting on the first : is enough for simple cases, but not sure about multilne support as I haven't implemented that.

This specific issue is discussed here, because several YAML parsers break on URLs https://stackoverflow.com/questions/11301650/how-to-escape-indicator-characters-i-e-or-in-yaml

**Comment by craignicol** [comment]
I had the same issue with the Typescript front matter parser, so I ended up with this simple implementation, which I'm sure has its own bugs

https://github.com/zettel-lint/zettel-lint/commit/e412f667b0bbe9a620ce7a35d9f40943b88fc69e

**Comment by vHanda** [comment]
I've thought about this a bit more, and this currently isn't a priority for me. I'm more focused on a deeper integration with Git and the desktop + web version.

If someone else could take a look at - https://github.com/GitJournal/yaml_serializer - I would appreciate it. I'll mark this issue accordingly.

**Comment by vHanda** [comment]
Also see - https://github.com/GitJournal/GitJournal/issues/16

**Comment by craignicol** [comment]
Additional things I've noticed : tags are removed if the line contains #

If value is empty, it's written as null when saved. I think this should remain empty, or be removed.
